### PR TITLE
create execWithTimeout function to execute command with specific timeout

### LINKED ISSUE #19: Make provider stops if the a task is running for too long
In that case, the task should just return whatever was already printed to stdout.

### PR DESCRIPTION
**Pull request by prin-r**
Fixes #19

### CODE REVIEW

File: chain/cmd/provider/main.go [hunk 1]
```diff
@@ -53,6 +55,20 @@ type Command struct {
 	Arguments []string `json:"args"`
 }
 
+func execWithTimeout(command Command, timeoutInMiliSecond int64) ([]byte, error) {
```
Comment (taobun) [inline chain/cmd/provider/main.go]: MilliSecond

File: chain/cmd/provider/main.go [hunk 2]
```diff
@@ -21,6 +23,7 @@ import (
 )
 
 var txSender cmtx.TxSender
+var allowedCommands = map[string]bool{"curl": true, "ping": true}
```
Comment (sorawit) [inline chain/cmd/provider/main.go]: Example of the workaround you need to do when a language is so stupid that it does not have `set`

File: chain/cmd/provider/main_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/bandprotocol/d3n/chain/x/zoracle"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	common "github.com/tendermint/tendermint/libs/common"
+	//sdk "github.com/cosmos/cosmos-sdk/types"
```
Comment (sorawit) [inline chain/cmd/provider/main_test.go]: Remove this?

**Review by sorawit — APPROVED** [review]
LGTM. @taobun any comment?

**Review by sorawit — CHANGES_REQUESTED** [review]
LGTM minus the minor comments. @taobun any comment?

File: chain/cmd/provider/main_test.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	// "github.com/stretchr/testify/require"
+)
+
+func TestExecWithTimeoutSuccess(t *testing.T) {
```
Comment (sorawit) [inline chain/cmd/provider/main_test.go:10]: We will see if all these tests cause the CI to be too slow. We can decide how to deal with that later if it ever becomes a problem.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/cmd/provider/main.go [hunk 5]
```diff
@@ -63,42 +80,39 @@ func handleRequest(event *abci.Event) {
...
 	for _, command := range commands {
-		if command.Cmd != "curl" {
-			fmt.Printf("handleRequest unknown command %s", command.Cmd)
-			return
+		if !allowedCommands[command.Cmd] {
+			return sdk.TxResponse{}, fmt.Errorf("handleRequest unknown command %s", command.Cmd)
 		}
-		cmd := exec.Command(command.Cmd, command.Arguments...)
-		query, err := cmd.Output()
+		query, err := execWithTimeout(command, 3000)
```
Comment (taobun) [inline chain/cmd/provider/main.go]: Bring 3000 as global variable scope.

**Review by taobun — CHANGES_REQUESTED** [review]

**Comment by sorawit** [comment]
@prin-r the CI is failing on `TestExecWithTimeoutSuccess`

File: chain/cmd/provider/main.go [hunk 6]
```diff
@@ -20,7 +22,10 @@ import (
 	"github.com/bandprotocol/d3n/chain/x/zoracle"
 )
 
+const LIMIT_TIMEOUT = 3 * time.Second
```
Comment (taobun) [inline chain/cmd/provider/main.go]: In go convention, private constant use lowerCamelCase. Therefore changed it to `limitTimeOut`

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]
lgtm

**Review by sorawit — APPROVED** [review]

**Pull request merged**